app/assets/javascripts/ckeditor/plugins/cortex_media_insert/plugin.js

```diff
@@ -0,0 +1,34 @@
+(function (global) {
+  'use strict';
+
+  global.CKEDITOR.plugins.add('cortex_media_insert', {
+    icons: 'media',
+    init: function (editor) {
+      editor.addCommand('insertMedia', {
+        exec: function (editor) {
+          window.MODALS.featured.open();
+
+          global.media_select = {};
+          global.media_select_defer = $.Deferred();
+          global.media_select_defer.promise(global.media_select);
+
+          global.media_select.done(function (media) {
```

Comment:
CKEditor doesn't exist in Cortex, it's loaded in and entirely handled within the scope of the plugin - we don't want to extend that into Cortex  because that would create an unbreakable attachment between the two, which goes against the overall design of it being a plugin to being with